compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -794,14 +780,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         let generic_params = self.arena.alloc_from_iter(generic_params);
         debug!(?generic_params);
 
-        if let Some(ctxt) = &mut self.captured_lifetimes {
-            ctxt.binders_to_ignore.insert(binder);
-        }
-        let ret = f(self, generic_params);
-        if let Some(ctxt) = &mut self.captured_lifetimes {
-            ctxt.binders_to_ignore.remove(&binder);
-        }
-        ret
+        f(self, generic_params)
```

Comment:
From what I can tell, this does not need to be a callback function anymore. It could just be (note the comment!)

```rust
/// Lowers a lifetime binder that defines `generic_params`, returning the corresponding HIR nodes.
/// The returned list includes any "extra" lifetime parameters that were added by the name resolver
/// owing to lifetime elision; this also populates the resolver's node-id->def-id map, so that later calls
/// to `opt_node_id_to_def_id` that refer to these extra lifetime parameters will be successful.
fn lower_lifetime_binder(
    &mut self,
    binder: NodeId,
    generic_params: &[GenericParam],
) -> &'hir [hir::GenericParam<'hir>] {
    ...
}
```

Reply:
Yes, this one was one of the things I wanted to do but forgot to address it yesterday :smile:. Fixed https://github.com/rust-lang/rust/pull/99867/commits/7506dfb13de10531215308629e723a2bca6bb929